<?xml version="1.0" ?>
<robot name="meskobot" xmlns:xacro="http://www.ros.org/wiki/xacro" >

    <xacro:property name="PI" value="3.14159"/>
    
    <ros2_control name="Mesko_bot" type="system">
      <hardware>
        <plugin>mesko_hardware/MeskoHardware</plugin>
        <param name="loop_rate">30</param>
        <param name="device">/dev/ttyUSB0</param>
        <param name="baud_rate">115200</param>
        <param name="timeout_ms">1000</param>
        
      </hardware>

        <joint name="Revolute 28">
            <command_interface name="position">
                <param name="min">-2.88</param>
                <param name="max">2.88</param>
            </command_interface>

    
            <state_interface name="position">
                  <param name="initial_value">0.0</param>
            </state_interface>

        </joint>
    
        <joint name="Revolute 29">
            <command_interface name="position">
                <param name="min">-2.18</param>
                <param name="max">2.18</param>
            </command_interface>
            
    
            <state_interface name="position">
                  <param name="initial_value">0.785</param>
            </state_interface>

        </joint>
    
        <joint name="Revolute 30">
            <command_interface name="position">
                <param name="min">-2.44</param>
                <param name="max">2.44</param>
            </command_interface>

            
            <state_interface name="position">
                  <param name="initial_value">-2.355</param>
            </state_interface>

        </joint>
    
        <joint name="Revolute 31">
            <command_interface name="position">
                <param name="min">-2.44</param>
                <param name="max">2.44</param>
            </command_interface>

    
            <state_interface name="position">
                  <param name="initial_value">1.57</param>
            </state_interface>

        </joint>
    
        <joint name="Revolute 32">
            <command_interface name="position">
                <param name="min">-2.44</param>
                <param name="max">2.44</param>
            </command_interface>

    
            <state_interface name="position">
                  <param name="initial_value">1.57</param>
            </state_interface>

        </joint>
	
	<joint name="Revolute 33">
            <command_interface name="position">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>

            <state_interface name="position">
                  <param name="initial_value">0.0</param>
            </state_interface>

        </joint>


    </ros2_control>

</robot>

